### ISSUE TITLE
All stations gone

### ISSUE THREAD

**Issue by JonnyTech** [y20k/transistor#294]
I use Transistor daily and currently running v4.0.5 on Android 9. Just started the app after rebooting my phone, to find all stations gone. Browsed to the settings folder, the collection folder is empty and the images folder has lots of guid-named subfolders which are all empty too. What is weird is that app settings are retained, ie it remembered dark mode. Are there any logs to help diagnose?

**Comment by y20k** [comment]
@JonnyTech Oh. I am sorry. That is a terrible bug. I am currently on vacation. I cannot investigate at the moment.

Just to make sure: In Transistor's file directory (within /Android/) was not a single file but only an images folder and its empty subfolders. 

I am not sure which subroutine of mine could cause such a mess. Maybe the system restore feature, but that would only work on phones using Play Services.

As to the logs question. Transistor does not keep logs. 

:unamused:

**Comment by JonnyTech** [comment]
Yes, all files gones, folder structure intact. No biggie, I have a backup, just wanted to report. Enjoy your holiday.

**Comment by nowster** [comment]
Just happened to me too after updating to Android 11.

I did a restore using Swift Backup back to a version from May. That worked.

Then I upgraded to the current version in Play Store and all the stations were gone.

**Comment by y20k** [comment]
@nowster 

I hope not to bother, but could you please post the exact order of steps you took? That might be of great help.

For example:
1. created backup of Transistor v3.2.4 using Swift Backup
2. updated to Transistor v4.0.x after that
3. stations were imported from old version
4. updated to Android 11
5. updated to Transistor v4.0.y
6. all station (that were once imported in step 3) are gone.

I know that recreating what happened exactly might be difficult. If you are not sure for some of the steps, that is totally okay and understandable. Anyway. An important start would be:

- Did the bug occur while updating from v4.0.x to v4.0.y?
- Or did the bug occur when updating from v3.2.4 to v4.0.x?

**Comment by nowster** [comment]
The fault happened when I was on Transistor 4.0.6 already. I updated to Android 11. All stations were gone after the Android update.

A fix:

- I totally uninstalled Transistor.
- I had an old backup of Transistor 3.2.4 in Swift Backup.
- I restored that version and its data.
- I opened it and played a radio station.
- I closed it and restored its data again using Swift Backup.
- I then upgraded to 4.0.6 from Play Store. It imported the stations correctly

However, v4.0.6 appears to be missing the feature to mark stations as favourites.

**Comment by y20k** [comment]
Okay. Thanks for clarifying.

> However, v4.0.6 appears to be missing the feature to mark stations as favourites.

This feature still works for me. But I somehow think that the import feature and #291 might be related. Does playback and changing stations work for you?

**Comment by nowster** [comment]
> This feature still works for me. But I somehow think that the import feature and #291 might be related. Does playback and changing stations work for you?

Playback and changing stations does work. So does adding stations to the home screen.

[Blush] I'd completely forgotten that making favourite is done by sliding. _Mea culpa!_ [/Blush]

Probably a completely separate issue: The "Currently Playing" stutters back to the start after only scrolling by about 3 characters if the sleep 💤 function is active.  My guess would be a scroll offset being reset to zero every 500ms (synchronous with the update of the countdown time).

**Comment by nowster** [comment]
And this morning I start Transistor and find it's spontaneously forgotten the list of stations that was working perfectly yesterday. No upgrades done. The only thing is that the phone had to be forcibly rebooted about 8 hours after I'd last used the Transistor app.

**Comment by y20k** [comment]
That may be a lead. I tried four scenarios (force reboot = long press power button):

1. forced reboot, while Transistor is visible in foreground and playing radio
2. forced reboot, while Transistor is visible in foreground and not playing radio
3. forced reboot, while Transistor is in background and playing radio
4. forced reboot, while Transistor is in background and not playing radio

In 1&2 I was able to reproduce the problem, which is a good start.

**Comment by y20k** [comment]
> In 1&2 I was able to reproduce the problem

And I added a small check (see [67b0152](https://github.com/y20k/transistor/commit/67b01528cebccf1526db1cabeb09296d471add6f)) that prevents writing a completely empty `collection.json` file, which might have happened, when I forced a device restart. At least it seems so. With the fix applied, I no longer could reproduce the problem.

**Comment by JonnyTech** [comment]
When it happened to me, there was no `collection.json` file present at all. Just an empty folder. The file was created upon adding a station. Maybe there are multiple issues at play here.

**Comment by peter-vanpoucke** [comment]
I seem to encounter this issue more often.

**Comment by y20k** [comment]
Hi, good evening @peter-vanpoucke 
Do you have any idea which action / interaction might be the cause?

**Comment by peter-vanpoucke** [comment]
Hi @y20k 

For now I don't. I tried force stopping it and checked after restarting the phone, but both didn't trigger it. I'm on stock Android 10.

**Comment by Licmeth** [comment]
I am also facing this issue regularly. In the past I had it always after updating Transistor app. From the current version (4.0.10) on, it also happens irregularly without any app update (or phone restarting). I don't know if this might have happened with earlier version too, as I started to use the app often just with this version.
Unfortunately, I also can not identify any use pattern that could cause the issue. Phone is a Xiami Mi A2 Lite with (vanilla) Android v10.
If I can support you with any action or info, please let me know!
